nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListenHTTP.java

```diff
@@ -351,6 +372,60 @@ synchronized private void createHttpServerFromService(final ProcessContext conte
         initialized.set(true);
     }
 
+    private ServerConnector createServerConnector(Server server, int port, SSLContextService sslContextService, boolean sslRequired, boolean needClientAuth) {
+        final SslContextFactory contextFactory = createSslContextFactory(sslContextService, sslRequired, needClientAuth);
+        return createServerConnector(server, port, contextFactory, sslRequired);
+    }
+
+    private ServerConnector createServerConnector(Server server, int port, SslContextFactory contextFactory, boolean sslRequired) {
+        final ServerConnector connector;
+        final HttpConfiguration httpConfiguration = new HttpConfiguration();
+        if (!sslRequired) {
+            // create the connector
+            connector = new ServerConnector(server, new HttpConnectionFactory(httpConfiguration));
+        } else {
+            // configure the ssl connector
+            httpConfiguration.setSecureScheme("https");
+            httpConfiguration.setSecurePort(port);
+            httpConfiguration.addCustomizer(new SecureRequestCustomizer());
+
+            // build the connector
+
+            connector = new ServerConnector(server, new SslConnectionFactory(contextFactory, "http/1.1"), new HttpConnectionFactory(httpConfiguration));
+        }
+
+        // configure the port
+        connector.setPort(port);
+        return connector;
+    }
+
+    private SslContextFactory createSslContextFactory(SSLContextService sslContextService, boolean sslRequired, boolean needClientAuth) {
+        final SslContextFactory contextFactory = new SslContextFactory.Server();
+        contextFactory.setNeedClientAuth(needClientAuth);
+
+        if (needClientAuth) {
+            contextFactory.setTrustStorePath(sslContextService.getTrustStoreFile());
+            contextFactory.setTrustStoreType(sslContextService.getTrustStoreType());
+            contextFactory.setTrustStorePassword(sslContextService.getTrustStorePassword());
+        }
+
+        if (sslRequired) {
+            final String keystorePassword = sslContextService.getKeyStorePassword();
+            final String keyStoreType = sslContextService.getKeyStoreType();
+            final String keyStorePath = sslContextService.getKeyStoreFile();
+
+            contextFactory.setKeyStorePath(keyStorePath);
+            contextFactory.setKeyManagerPassword(keystorePassword);
```

Comment:
Although this simply moves the existing implementation to a new method, this would be a good opportunity to set the keyManagerPassword from the getKeyPassword() method on SSLContextService, when configured.

Reply:
I believe that might break existing setups. If it does need to be modified, it is reasonable to open a separate Jira ticket for this modification so that it is not "shipped" as a side-effect of an enhancement.